keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -71,255 +73,287 @@ private void verifyEnvironmentVariables() {
     assertThat(failEntry).isEmpty();
+
+    deleteSecrets();
   }
 
   @Test
   void throwsAwayObjectsWhichMapToNull() {
-    createSecret(true, false, false);
-
-    final String nullEntryName = secretNames.get(1);
+    createSecret(false, false);
+    createSecret(false, false);
 
+    final String nullEntryName = testSecretNames.get(1);
     Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
         awsSecretsManagerExplicit.mapSecrets(
-            secretTags.keySet().stream().collect(Collectors.toList()),
-            secretTags.values().stream().collect(Collectors.toList()),
+            testSecretTags.keySet(),
+            testSecretTags.values(),
             (name, value) -> {
               if (name.equals(nullEntryName)) {
                 return null;
               }
               return new AbstractMap.SimpleEntry<>(name, value);
             });
 
-    validateMappedSecret(secretEntries, secretNames.get(0));
-
+    validateMappedSecret(secretEntries, testSecretNames.get(0));
     final Optional<AbstractMap.SimpleEntry<String, String>> nullEntry =
         secretEntries.stream().filter(e -> e.getKey().equals("MyBls")).findAny();
     assertThat(nullEntry).isEmpty();
+
+    deleteSecrets();
+  }
+
+  private static void initAwsSecretsManagers() {
+    awsSecretsManagerDefault = AwsSecretsManager.createAwsSecretsManager();
+    awsSecretsManagerExplicit =
+        AwsSecretsManager.createAwsSecretsManager(
+            RO_AWS_ACCESS_KEY_ID, RO_AWS_SECRET_ACCESS_KEY, AWS_REGION);
+    awsSecretsManagerInvalidCredentials =
+        AwsSecretsManager.createAwsSecretsManager("invalid", "invalid", AWS_REGION);
+  }
+
+  private static void initTestSecretsManager() {
+    awsBasicCredentials =
+        AwsBasicCredentials.create(RW_AWS_ACCESS_KEY_ID, RW_AWS_SECRET_ACCESS_KEY);
+    credentialsProvider = StaticCredentialsProvider.create(awsBasicCredentials);
+    testSecretsManagerClient =
+        SecretsManagerClient.builder()
+            .credentialsProvider(credentialsProvider)
+            .region(Region.of(AWS_REGION))
+            .build();
+  }
+
+  private static void initTestVariables() {
+    testSecretNames = new ArrayList<>();
+    testSecretTags = new HashMap<>();
+  }
+
+  private static void closeTestSecretsManager() {
+    testSecretsManagerClient.close();
+  }
+
+  private static void closeAwsSecretsManagers() {
+    awsSecretsManagerDefault.close();
+    awsSecretsManagerExplicit.close();
+    awsSecretsManagerInvalidCredentials.close();
+  }
+
+  private static void closeClients() {
+    closeAwsSecretsManagers();
+    closeTestSecretsManager();
+  }
+
+  private CreateSecretResponse createSecret(final boolean multipleTags, final boolean sharedTag) {
+    testSecretNamePrefix = "signers-aws-integration/";
+    testSecretName = testSecretNamePrefix + UUID.randomUUID();
+    testSecretNames.add(testSecretName);
+
+    final List<Tag> tags = new ArrayList<>();
+
+    if (sharedTag) {
```

Comment:
Think it would be clearer to just pass in any tags to associated with the secrets